aws/vpc_flow_log_monitoring/lambda_function.py

```diff
@@ -23,27 +21,64 @@
 
 DD_SITE = os.getenv("DD_SITE", default="datadoghq.com")
 
-# retrieve datadog options from KMS
-KMS_ENCRYPTED_KEYS = os.environ['kmsEncryptedKeys']
-kms = boto3.client('kms')
-
-try:
-    decrypted = kms.decrypt(
-        CiphertextBlob=b64decode(KMS_ENCRYPTED_KEYS),
-        EncryptionContext={'LambdaFunctionName': os.environ['AWS_LAMBDA_FUNCTION_NAME']},
-    )['Plaintext']
-except botocore.exceptions.ClientError:
-    decrypted = kms.decrypt(
-        CiphertextBlob=b64decode(KMS_ENCRYPTED_KEYS),
-    )['Plaintext']
-
-datadog_keys = json.loads(decrypted)
-
+def _datadog_keys():
+    if 'kmsEncryptedKeys' in os.environ:
+        KMS_ENCRYPTED_KEYS = os.environ['kmsEncryptedKeys']
+        kms = boto3.client('kms')
+        # kmsEncryptedKeys should be created through the Lambda's encryption
+        # helpers and as such will have the EncryptionContext
+        return json.loads(kms.decrypt(
+            CiphertextBlob=base64.b64decode(KMS_ENCRYPTED_KEYS),
+            EncryptionContext={'LambdaFunctionName': os.environ['AWS_LAMBDA_FUNCTION_NAME']},
+        )['Plaintext'])
+
+    if 'DD_API_KEY_SECRET_ARN' in os.environ:
+        SECRET_ARN = os.environ['DD_API_KEY_SECRET_ARN']
+        DD_API_KEY = boto3.client('secretsmanager').get_secret_value(SecretId=SECRET_ARN)['SecretString']
+        return {'api_key': DD_API_KEY}
+
+    if 'DD_API_KEY_SSM_NAME' in os.environ:
+        SECRET_NAME = os.environ['DD_API_KEY_SSM_NAME']
+        DD_API_KEY = boto3.client('ssm').get_parameter(
+            Name=SECRET_NAME, WithDecryption=True
+        )['Parameter']['Value']
+        return {'api_key': DD_API_KEY}
+
+    if 'DD_KMS_API_KEY' in os.environ:
+        ENCRYPTED = os.environ['DD_KMS_API_KEY']
+
+        # For interop with other DD Lambdas taking in DD_KMS_API_KEY, we'll
+        # optionally try the EncryptionContext associated with this Lambda.
+        try:
+            DD_API_KEY = boto3.client('kms').decrypt(
+                CiphertextBlob=base64.b64decode(ENCRYPTED),
+                EncryptionContext={'LambdaFunctionName': os.environ['AWS_LAMBDA_FUNCTION_NAME']},
+            )['Plaintext']
+        except botocore.exceptions.ClientError:
+            DD_API_KEY = boto3.client('kms').decrypt(
+                CiphertextBlob=base64.b64decode(ENCRYPTED),
+            )['Plaintext']
+
+        if type(DD_API_KEY) is bytes:
+            DD_API_KEY = DD_API_KEY.decode('utf-8')
+        return {'api_key': DD_API_KEY}
+
+    if 'DD_API_KEY' in os.environ:
+        DD_API_KEY = os.environ['DD_API_KEY']
+        return {'api_key': DD_API_KEY}
+
+    raise ValueError("Datadog API key is not defined, see documentation for environment variable options")
+
+
+# Preload the keys so we can bail out early if they're misconfigured
 # Alternatively set datadog keys directly
 # datadog_keys = {
 #     "api_key": "abcd",
 #     "app_key": "efgh",
 # }
+datadog_keys = _datadog_keys()
+print('INFO Lambda function initialized, ready to send metrics')
```

Comment:
Why not use python builtin logger?

Reply:
No reason. I can change the prints to be info logs.